src/zarr/testing/store.py

```diff
@@ -0,0 +1,81 @@
+import pytest
```

Comment:
importing `zarr.testing.store` will raise an `ImportError` if pytest is not installed. I think that could be fine but calling it out so others have a chance to comment. I think there are some things we can do to further hide this import if folks object to the current configuration.